sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpConnectionScope.cs

```diff
@@ -168,37 +170,42 @@ public override bool IsDisposed
 
         private readonly object _syncLock = new();
         private readonly TimeSpan _operationTimeout;
+        private readonly ServiceBusClient _client;
```

Comment:
I'm not a fan of inverting the dependency flow here.   My goal was to have awareness flow in one direction - the client layer was aware of the transport layer but not vice-versa.   This makes it circular.    

Have we considered putting the connected/disconnected events on the connection scope and surfacing it through the transport client?    That would add a small bit more plumbing, but keep the dependency flow in a single direction.

Reply:
Yes, that is a fair point - was trying to just get things working quickly to validate the overall approach.